garak/analyze/deep_analysis/deep_analysis.py

```diff
@@ -0,0 +1,126 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Provide feedback, recommendations, and qualitative feedback on scan results.
+"""
+
+import json
+from multiprocessing import Pool
+from functools import lru_cache
+import pandas as pd
+import numpy as np
+from pathlib import Path
+from typing import Tuple
+from analytics import *
+
+from garak.data import path as data_path
+
+
+ANALYSIS_FILE = data_path / "deep_analysis" / "deep_analysis.csv"
+
+
+@lru_cache
+def load_scores(filepath: Path) -> pd.DataFrame:
+    df = pd.read_csv(filepath)
+    return df
+
+
+def get_position(probe_name: str, score: float, filepath: Path) -> int:
+    """
+    Get the position of a target's probe score in relation to other models in the bag
+
+    Parameters
+    ----------
+    probe_name: str: Name of the probe
+    score: float: Value of the score
+    filepath: Path: Path to file containing the values of models in the bag
+
+    Returns
+    -------
+    position: int: The position of the model in the set of sorted scores.
+    """
+    scores = load_scores(filepath)
+    probe_scores = np.sort(scores[probe_name].to_numpy())
+    position = int(np.where(probe_scores <= score)[0])
+    return position
+
+
+def tier_1(analysis_dict: dict) -> str:
+    # Jailbreak analysis
+    jailbreak_analysis = jailbreak(analysis_dict["jailbreak"])
+    # Other analyses, once available
+    # Concatenate all the feedback together with newlines.
+    feedback_string = "\n".join([jailbreak_analysis])
+
+    return feedback_string
+
+
+def tier_2(analysis_dict: dict) -> str:
+    pass
+
+
+def deep_analysis(report_path, bag_path=ANALYSIS_FILE) -> Tuple[str, str]:
+    """
+    Take garak report jsonl file and perform qualitative analysis on the probe results for the target.
```

Comment:
`perform qualitative analysis`

It's all quantitative right? We compare quantities and choose blocks of text, no human in the loop, no qualitative method